### ISSUE TITLE
Native build broken with main due to OpenTracing 0.33.0 upgrade

### ISSUE THREAD

**Issue by gsmet** [quarkiverse/quarkus-datadog-opentracing#67]
The native build with main got recently broken:
```
com.oracle.svm.core.util.UserError$UserException: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: io.opentracing.Scope.span(). To diagnose the issue you can use the --allow-incomplete-classpath option. The missing method is then reported at run time when it is accessed the first time.
Detailed message:
Trace: 
	at parsing datadog.opentracing.TypeConverter$CustomScope.span(TypeConverter.java:161)
Call path from entry point to datadog.opentracing.TypeConverter$CustomScope.span(): 
	at datadog.opentracing.TypeConverter$CustomScope.span(TypeConverter.java:161)
	at datadog.opentracing.OTScopeManager$OTScope.close(OTScopeManager.java:69)
	at com.oracle.svm.core.posix.NativeSecureRandomFilesCloserShutdownHook.close(PosixSunSecuritySubstitutions.java:86)
```

The issue is that the Datadog OpenTracing library (`dd-trace-ot`) is still depending on a method `io.opentracing.Scope.span()` that has been removed in OpenTracing API 0.33.0. I don't see a lot of things we can do to fix this except bugging Datadog to upgrade to 0.33.0 and adjust.

@nvivot Do you have any relations with Datadog or know where we could ask?

**Comment by nicolas-vivot** [comment]
Hi @gsmet 

Yep i've seen that before going on weekend.

I will submit a request for that and pass through my company support to request them to work on it asap.

**Comment by nicolas-vivot** [comment]
[Opened issue](https://github.com/DataDog/dd-trace-java/issues/3293) on their repo.

**Comment by devinsba** [comment]
How would I reproduce this issue from this repo's point of view. I'm trying to understand how the `CustomScope` class is getting invoked, normally it shouldn't be invoked at all so that caught my attention

**Comment by nicolas-vivot** [comment]
@devinsba 

The issue here is not at runtime but at compile time.
Not sure if you are familiar with Quarkus, but in native mode it runs on GraalVM.
The build of this extension is currently failing (https://github.com/quarkiverse/quarkus-datadog-opentracing/runs/4614614894?check_suite_focus=true) when compiling the extension in native to be able to run the integration tests in native mode.

During the compilation, there is a scan phase and graal reports that it find the use of an unknown class and therefore report it and stop the compilation.
At this point, there are two options:

- either we fix the source code (the reason why i opened the issue on Datadog since it's your source code)
- either we use an option in this extension to tell to graal to ignore such unknown class discovery and report it at runtime instead of at build time

Option 1 is by far the best (and should be the priority), but option 2 can be used to unlock the situation. The problem i have with option 2 is that it would potentially break at runtime which is really bad thing.

Now, if you says that this CustomScope should not be used:

- why is it here ? what for ? (just for my understanding of DD lib)
- is there a way to quick fix this one only since it appears to be only that class that is affected?

Here is what i will do (while waiting for your answer / investigation):
- i will add an option to ignore & report at runtime such issue and build the extension with the latest version of Quarkus
- test the updated extension on one of my service and run load tests on it to see whether this would have an impact at runtime

**Comment by nicolas-vivot** [comment]
So, i have performed some functional and load tests within an application and there does not seems to have any runtime issue while ignoring this reported issue at build time.
But as i explained, this will force users to add the `--allow-incomplete-classpath` option to their builds, which is not great at all.

@gsmet 
For now, i will add the `--allow-incomplete-classpath` to this extension integration tests configuration to avoid the build to fail for that reason and will follow the fix on Datadog.

I will also add some notes on the documentation to warn about this.

@devinsba 
Would be nice if you can find a way to patch the CustomScope inner class to comply with version 0.33.0 without upgrading.
I will keep this issue opened until it's fixed on datadog side.

**Comment by nicolas-vivot** [comment]
Another option and probably safer than the `--allow-incomplete-classpath` option is to downgrade the opentracing library version to 0.32.0 in the applications.

Both temporary solution are provided on the release note here:
https://github.com/quarkiverse/quarkus-datadog-opentracing/releases/tag/1.1.3

**Comment by nicolas-vivot** [comment]
The following pull request will fix the issue on datadog library

https://github.com/DataDog/dd-trace-java/pull/3299

Release 0.93.0 should then be compatible with the latest versions of Quarkus.

**Comment by gsmet** [comment]
Thanks for following up :).

**Comment by nicolas-vivot** [comment]
Fixed with release [1.1.4](https://github.com/quarkiverse/quarkus-datadog-opentracing/releases/tag/1.1.4)
